File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int answer = 0;

        Arrays.sort(nums);
        int max = 0;
        int cnt = 1;
        int prev = nums[0];
        for(int i=1;i<nums.length;i++) {
            if(prev == nums[i]) cnt++;
            else {
                if(max < cnt) {
                    answer = nums[i-1];
                    max = cnt;
                }
                prev = nums[i];
                cnt = 1;
            }
        }
        if(max < cnt) answer = nums[nums.length-1];
        return answer;
    }
}

File: 0169-majority-element/0169-majority-element.py
from collections import Counter

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        res = 0
        
        mj_cnt = len(nums) // 2
        count_nums = Counter(nums)
        for num, cnt in count_nums.most_common():
            if cnt > mj_cnt: 
                return num
        